pkg/settlement/swap/swapprotocol/swapprotocol.go

```diff
@@ -186,8 +186,7 @@ func (s *Service) EmitCheque(ctx context.Context, peer swarm.Address, cheque *ch
 		if err != nil {
 			_ = stream.Reset()
 		} else {
-			// don't wait for full close to avoid deadlocks if cheques are sent simultaneously in both directions
-			go stream.FullClose()
```

Comment:
why remove the goroutine

Reply:
because we want to wait for the stream to be closed. this gives us at least some basic assurance that the counterparty has had time to process the cheque. in the meantime settlement is still holding that peers accounting lock, therefore preventing overspending.